### PR TITLE
consul-template: update to get it functional

### PR DESCRIPTION
**Pull request by lpaulmp**
Fix the syntax of the check template #114, and fix consul reload adding a `requires` in the service file #102 .

Apply the formula `salt-call --local state.sls consul.template-tool` finished successful 
```
local:
----------
          ID: unzip
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 03:31:46.922605
    Duration: 324.173 ms
     Changes:   
----------
          ID: consul-template-archive
    Function: archive.extracted
        Name: /usr/local/bin/consul-template-0.19.4
      Result: True
     Comment: Path /usr/local/bin/consul-template-0.19.4/consul-template exists
     Started: 03:31:47.250927
    Duration: 0.561 ms
     Changes:   
----------
          ID: consul-template-archive
    Function: file.directory
        Name: /usr/local/bin/consul-template-0.19.4
      Result: True
     Comment: The directory /usr/local/bin/consul-template-0.19.4 is in the correct state
     Started: 03:31:47.251681
    Duration: 0.865 ms
     Changes:   
----------
          ID: consul-template-bin
    Function: file.symlink
        Name: /usr/local/bin/consul-template
      Result: True
     Comment: Symlink /usr/local/bin/consul-template is present and owned by root:root
     Started: 03:31:47.252774
    Duration: 1.293 ms
     Changes:   
----------
          ID: consul-user
    Function: group.present
        Name: consul
      Result: True
     Comment: Group consul is present and up to date
     Started: 03:31:47.254542
    Duration: 0.426 ms
     Changes:   
----------
          ID: consul-user
    Function: user.present
        Name: consul
      Result: True
     Comment: User consul is present and up to date
     Started: 03:31:47.255934
    Duration: 1.095 ms
     Changes:   
----------
          ID: consul-user
    Function: file.directory
        Name: /home/consul
      Result: True
     Comment: The directory /home/consul is in the correct state
     Started: 03:31:47.257225
    Duration: 2.649 ms
     Changes:   
----------
          ID: consul-tpl-conf-path
    Function: file.directory
        Name: /home/consul/template-tool
      Result: True
     Comment: The directory /home/consul/template-tool is in the correct state
     Started: 03:31:47.260169
    Duration: 0.657 ms
     Changes:   
----------
          ID: consul-tpl-base-config
    Function: file.managed
        Name: /home/consul/template-tool/00-consul-template.hcl
      Result: True
     Comment: File /home/consul/template-tool/00-consul-template.hcl is in the correct state
     Started: 03:31:47.261277
    Duration: 53.116 ms
     Changes:   
----------
          ID: consul-tpl-consul-service
    Function: file.managed
        Name: /home/consul/conf.d/consul_template_service.json
      Result: True
     Comment: File /home/consul/conf.d/consul_template_service.json is in the correct state
     Started: 03:31:47.316576
    Duration: 10.369 ms
     Changes:   
----------
          ID: consul-service-check-reload
    Function: cmd.run
        Name: consul reload
      Result: True
     Comment: Command "consul reload" run
     Started: 03:31:47.327335
    Duration: 36.127 ms
     Changes:   
              ----------
              pid:
                  5120
              retcode:
                  0
              stderr:
              stdout:
                  Configuration reload triggered
----------
          ID: consul-tpl-service
    Function: file.managed
        Name: /etc/systemd/system/consul-template.service
      Result: True
     Comment: File /etc/systemd/system/consul-template.service is in the correct state
     Started: 03:31:47.363761
    Duration: 35.097 ms
     Changes:   
----------
          ID: consul-tpl-templates-path
    Function: file.directory
        Name: /srv/consul-templates
      Result: True
     Comment: The directory /srv/consul-templates is in the correct state
     Started: 03:31:47.401199
    Duration: 0.921 ms
     Changes:   
----------
          ID: consul-tpl-service
    Function: service.running
        Name: consul-template
      Result: True
     Comment: The service consul-template is already running
     Started: 03:31:47.402480
    Duration: 39.002 ms
     Changes:   

Summary for local
-------------
Succeeded: 14 (changed=1)
Failed:     0
-------------
Total states run:     14
Total run time:  506.351 ms
```

### CODE REVIEW

File: consul/template-tool/files/base_config.hcl [hunk 1]
```diff
@@ -9,12 +9,13 @@
 {%- set vault = salt['pillar.get']('consul_template:vault', False) %}
 {%- set vault_url = salt['pillar.get']('consul_template:vault_url', 'https://vault.service.consul:8200') %}
 {%- set vault_token = salt['pillar.get']('consul_template:vault_token', 'VAULT_TOKEN') %}
+
 consul = "{{ consul_addr }}"
-token = "{{ consul_token }}" {# May also use the env var CONSUL_TOKEN #}
-retry = "10s"
+#token = "{{ consul_token }}" {# May also use the env var CONSUL_TOKEN #}
+#retry = "10s" {# doesn't work as HCL file format TODO: try as json file #}
```
Comment (ketzacoatl) [inline consul/template-tool/files/base_config.hcl]: where possible, we want to stick with the HCL format (over JSON). Has the `retry` parameter been deprecated or renamed?

File: consul/template-tool/files/base_config.hcl [hunk 2]
```diff
@@ -9,12 +9,13 @@
 {%- set vault = salt['pillar.get']('consul_template:vault', False) %}
 {%- set vault_url = salt['pillar.get']('consul_template:vault_url', 'https://vault.service.consul:8200') %}
 {%- set vault_token = salt['pillar.get']('consul_template:vault_token', 'VAULT_TOKEN') %}
+
 consul = "{{ consul_addr }}"
-token = "{{ consul_token }}" {# May also use the env var CONSUL_TOKEN #}
-retry = "10s"
+#token = "{{ consul_token }}" {# May also use the env var CONSUL_TOKEN #}
```
Comment (ketzacoatl) [inline consul/template-tool/files/base_config.hcl]: I'm not sure I understand, why drop `token`?

**Comment by ketzacoatl** [comment]
Great job figuring out the updates included in the PR, I have only a couple small questions.

File: consul/template-tool/files/base_config.hcl (hunk 1 above)
Comment (lpaulmp) [reply consul/template-tool/files/base_config.hcl]: what do you think open an issue to address this specific case? this wasn't the root of the problem that prevent initialize the monitor for consul-template

Comment (ketzacoatl) [reply consul/template-tool/files/base_config.hcl]: OK, so the config needs to be fixed and you will do it in a separate PR, I follow. Thanks for explaining.

**Pull request merged**